### PR TITLE
Make milling results definable in JSON

### PR DESCRIPTION
**Pull request by BevapDin**
```SUMMARY: Infrastructure "Milling products can be defined in JSON"```

- Removes the (undocumented) flag "MILLABLE".

- Adds an item slot that defines what an item can be milled into (e.g. for "wheat" that would be "flour").

- Fixes mills to produces flour at the same conversion rate as manual crafting. Mills used to give 15 flour for 1 input item. But crafting it with a quern requires different amounts of input items to get the same 15 flour as result. For example it takes **four** oats to craft 15 flour. This has been fixed (80 oats give only 300 flout, 80 wheat give 1200 flour).

- Fixes the product to have the same properties as if crafted by hand (inheriting flags and components). This makes flour milled from oats contain different vitamins than when milled from wheat, just like you would get if you had crafted it manually.

- Makes chitin pieces and bone (human as well) millable. The result is chitin powder and bone meal.

- Fixes a crash (see commit).

--

Some remarks:

Why are food items defined in a folder "data/json/items/comestibles/", but the recipes for them are located in "data/json/recipes/food/" ("food" vs "comestibles")?

Cooked acorns, dried rice and starch can be converted to flour with a quern, but not with a mill (existing behaviour). Should this be possible?

### CODE REVIEW

File: doc/JSON_INFO.md [hunk 1]
```diff
@@ -1678,6 +1678,10 @@ See also VEHICLE_JSON.md
     [ "9mm", [ "glockmag" ] ]                // The first magazine specified for each ammo type is the default
     [ "45", [ "m1911mag", "m1911bigmag" ] ],
 ],
+"milling": {                                 // Optional. If given, the item can be milled in a water/wind mill.
+  "into": "flour",                           // The item id of the resul of the milling.
```
Comment (AMurkin) [inline doc/JSON_INFO.md:1682]: Typo: result.

File: src/iexamine.cpp [hunk 2]
```diff
@@ -214,6 +213,9 @@ static const std::string flag_VARSIZE( "VARSIZE" );
 static const std::string flag_WALL( "WALL" );
 static const std::string flag_WRITE_MESSAGE( "WRITE_MESSAGE" );
 
+// @TODO maybe make this a property of the item (depend on volume/type)
+static const auto milling_time = 6_hours;
```
Comment (anothersimulacrum) [inline src/iexamine.cpp:217]:
```suggestion
static const time_duration milling_time = 6_hours;
```

File: src/iexamine.cpp [hunk 3]
```diff
@@ -4969,9 +4964,18 @@ void iexamine::mill_finalize( player &, const tripoint &examp, const time_point
     }
 
     for( item &it : items ) {
-        if( it.has_flag( flag_MILLABLE ) && it.get_comestible() ) {
-            it.calc_rot_while_processing( 6_hours );
-            item result( "flour", start_time + 6_hours, it.charges * 15 );
+        if( it.type->milling_data ) {
+            it.calc_rot_while_processing( milling_time );
+            const auto &mdata = *it.type->milling_data;
```
Comment (anothersimulacrum) [inline src/iexamine.cpp:4969]:
```suggestion
            const islot_milling &mdata = *it.type->milling_data;
```

File: doc/JSON_INFO.md (hunk 1 above)
Comment (ZhilkinSerg) [inline doc/JSON_INFO.md:1682]:
```suggestion
  "into": "flour",                           // The item id of the result of the milling.
```

**Pull request merged**